src/util/HaltingAnalysis.ts

```diff
@@ -29,10 +30,8 @@ function isQuorumSet(n: string | AnalysisQuorumSet): n is AnalysisQuorumSet {
   return (n as AnalysisQuorumSet).threshold !== undefined;
 }
 
-function isNested(
-  set: string[] | NetworkQuorumSet[]
-): set is NetworkQuorumSet[] {
-  return typeof set[0] != "string";
+function isNested(set: string | NetworkQuorumSet): set is NetworkQuorumSet {
```

Comment:
changed to check each individual item in the list, vs checking the overall list and assuming they're all the same type